week-3/implement/is-valid-triangle.js

```diff
@@ -1,4 +1,15 @@
 // Implement a function isValidTriangle
+function isValidTriangle(a, b, c) {
+  if (a <= 0 && b <= 0 && c <= 0) {
+    return false;
+  } else {
+    if (a + b > c && b + c > a && c + a > b) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+}
```

Comment:
This will definitely work! But again it may be improved:

1) `a <= 0 && b <= 0 && c <= 0` - this check means that if all 3 sides are equal to 0, then it's not a triangle. But in fact it's not a triangle if either of the sides is equal to 0. Either `a`, or `b`, or `c`.

2) As for the rest of this function, could you try refactor it yourself using my advice from `is-proper-fraction.js` task?

Reply:
so it's gonna be like
```
function isValidTriangle(a, b, c) {
  if (a <= 0 && b <= 0 && c <= 0) {
      return false;
    }
  return (a + b > c && b + c > a && c + a > b)
}
```

Reply:
What I actually meant was using `||` instead of `&&` in the first `if` check:
```
  if (a <= 0 || b <= 0 || c <= 0) {
    return false;
  } else {
    return (a + b > c && b + c > a && c + a > b); // for this line your refactoring is perfect!
```
or... we can even make it shorter:
```
return !(a <= 0 || b <= 0 || c <= 0) || (a + b > c && b + c > a && c + a > b);
```
but the last may be not very clear.